src/browser/standalone.ts

```diff
@@ -0,0 +1,128 @@
+import { Config } from "../config";
+import { NewBrowser } from "./new-browser";
+import { ExistingBrowser } from "./existing-browser";
+import { Calibrator } from "./calibrator";
+import { AsyncEmitter } from "../events";
+import { BrowserName, type W3CBrowserName } from "./types";
+import { getNormalizedBrowserName } from "../utils/browser";
+import { LOCAL_GRID_URL } from "../constants/config";
+import { WebdriverPool } from "../browser-pool/webdriver-pool";
+import RuntimeConfig from "../config/runtime-config";
+import type { CommonConfig } from "../config/types";
+
+export type StandaloneBrowserOptions = Pick<
+    CommonConfig,
+    | "automationProtocol"
+    | "desiredCapabilities"
+    | "gridUrl"
+    | "baseUrl"
+    | "httpTimeout"
+    | "pageLoadTimeout"
+    | "waitTimeout"
+    | "waitInterval"
+    | "headless"
+    | "prepareBrowser"
+    | "windowSize"
+    | "orientation"
+    | "headers"
+    | "strictSSL"
+    | "user"
+    | "key"
+    | "system"
+>;
+
+export type StandaloneBrowserOptionsInput = Partial<StandaloneBrowserOptions>;
+
+export async function launchBrowser(options: StandaloneBrowserOptionsInput = {}): Promise<WebdriverIO.Browser> {
+    const desiredCapabilities = options.desiredCapabilities || {};
+
+    const browserName = desiredCapabilities.browserName || BrowserName.CHROME;
+    const normalizedBrowserName = getNormalizedBrowserName(browserName) as W3CBrowserName;
+
+    if (!normalizedBrowserName) {
+        throw new Error(
+            [
+                `Running browser "${browserName}" is unsupported`,
+                `Supported browsers: "chrome", "firefox", "safari", "edge"`,
+            ].join("\n"),
+        );
+    }
+
+    RuntimeConfig.getInstance().extend({
+        debug: options.system?.debug ?? true,
+        local: true,
+    });
+
+    const browserConfig = {
+        desiredCapabilities: {
+            browserName,
+            ...desiredCapabilities,
+        },
+        gridUrl: options.gridUrl || LOCAL_GRID_URL,
+        baseUrl: options.baseUrl,
+        headless: options.headless !== undefined ? options.headless : true,
+        pageLoadTimeout: options.pageLoadTimeout,
+        httpTimeout: options.httpTimeout,
+        waitTimeout: options.waitTimeout,
+        waitInterval: options.waitInterval,
+        automationProtocol: options.automationProtocol,
+        windowSize: options.windowSize,
+        orientation: options.orientation,
+        headers: options.headers,
+        strictSSL: options.strictSSL,
+        user: options.user,
+        key: options.key,
+        prepareBrowser: options.prepareBrowser,
+    };
+
+    const config = new Config({
+        browsers: {
+            [browserName]: browserConfig,
+        },
+        system: options.system || {
+            debug: true,
+        },
+    });
+
+    const webdriverPool = new WebdriverPool();
```

Comment:
Didn't you consider to make it singleton?

Reply:
Fair, made it so that it was created only once.